### PR TITLE
Skipping ensure_page_safe on timeout. 

### PR DESCRIPTION
**Pull request by lhellebr**
Workaround for e.g. https://bugzilla.redhat.com/show_bug.cgi?id=2106022

### CODE REVIEW

File: airgun/browser.py [hunk 1]
```diff
@@ -236,11 +237,20 @@ class AirgunBrowserPlugin(DefaultPlugin):
...
+        If self.success_on_timeout is True, the function doesn't raise an exception
+        and continues as if the page was safe instead. This can be used to bypass
+        some bugs, e.g. https://bugzilla.redhat.com/show_bug.cgi?id=2106022
         """
-        super().ensure_page_safe(timeout)
+        try:
+            super().ensure_page_safe()
+        except TimedOutError as e:
+            if not self.success_on_timeout:
+                raise e
```
Comment (mshriver) [inline airgun/browser.py:253]: No need to capture the exception object here, you can drop `as e` and `raise e`, just calling `raise` will propagate the exception object.

**Comment by lhellebr** [comment]
I wanted to be simple here because I don't see this being used often but you're right, let's make things properly. I'll change the PR alongside your example.

**Comment by lhellebr** [comment]
Fixed, the test run failed with an unrelated error.
`selenium.common.exceptions.NoSuchElementException: Message: Could not find an element Locator(by='xpath', locator="//div[@id='scsi_controllers']")`

**Review by mshriver — APPROVED** [review]
Looks great! Thanks for taking the time to make this enhancement instead of working around in the impacted test only.

**Comment by latran** [comment]
@lhellebr Can you provide a test case that causes the failure?

**Comment by lhellebr** [comment]
`tests/foreman/ui/test_computeresource_vmware.py::test_positive_access_vmware_with_custom_profile`
Perhaps others but this is my reproducer

**Review by latran — APPROVED** [review]
acking.  I will investigate your unrelated error and see what's going on.

**Comment by lhellebr** [comment]
> acking. I will investigate your unrelated error and see what's going on.

Don't, I'm working on it right now. Some locator changes.

File: airgun/browser.py [hunk 2]
```diff
@@ -236,11 +238,30 @@ class AirgunBrowserPlugin(DefaultPlugin):
...
     def ensure_page_safe(self, timeout='30s'):
         """Ensures page is fully loaded.
         Default timeout was 10s, this changes it to 30s.
+        If self.ignore_ensure_page_safe_timeout is True, the function doesn't raise an exception
+        and continues as if the page was safe instead. This can be used to bypass
+        some bugs, e.g. https://bugzilla.redhat.com/show_bug.cgi?id=2106022
         """
-        super().ensure_page_safe(timeout)
+        try:
+            super().ensure_page_safe()
```
Comment (lhellebr) [inline airgun/browser.py:261]: I somehow dropped the timeout parameter from a function all. This will need to be fixed in a separate PR.
